design/api-specification-design.md

```diff
@@ -0,0 +1,86 @@
+# api-specification-design
+Personal notes about REST API design [here](https://github.com/perrymant/learning-notes/blob/master/17-rest-api-design.md)
+
+## 1. report
+The `report` request obtains a report of their account.
+
+#### Request
+| Method | URL         |
+|--------|-------------|
+| GET    | api/report  |
+
+#### Response
+- 200: a `report statement` will be sent:
+
+```json
+{
+    "data" : {
+        "report" : "---report statement-(see below)---"
+     }
+}
+```
+
+An example `report statement` for status code 200 :
+```
+{
+"report": [
+  {
+    "date": ...,
```

Comment:
We're missing the good bits!
What's the format of the date? ISO8601 is always a good choice.
How is the amount and balance given? In pence or as a decimal? Or another format entirely?